locations/storefinders/yext.py

```diff
@@ -84,3 +86,24 @@ def parse(self, response):
 
     def parse_item(self, item, location, **kwargs):
         yield item
+
+    def storefinder_exists(response: Response) -> bool | Request:
+        # "https://cdn.yextapis.com/v2/accounts/me/entities?api_key=
+        if response.xpath('//script[contains(text(), "window.Yext")]').get():
+            return True
+
+        # https://locations.cariboucoffee.com/
+        if response.xpath('//script[contains(text(), "locator-google")]').get():
+            return True
+
+        # https://www.fnb-online.com/atm-branch-locator
+        # As per https://hitchhikers.yext.com/guides/add-searchbar-guide/01-intializethelibrary/
+        if response.xpath('//script[contains(@src, "answers-search-bar")]').get():
+            return True
+
+        return False
+
+    def extract_spider_attributes(response: Response) -> dict | Request:
+        return {
+            "allowed_domains": [urlparse(response.url).netloc],
```

Comment:
I see Yext referenced a lot on various brand webpages, but not always for a store locator. In many cases there is reference to Yext libraries, but the store finder doesn't make a call to the expected Yext store locator API endpoint of:

`https://cdn.yextapis.com/v2/accounts/me/entities?api_key={self.api_key}&v={self.api_version}&limit={self.page_limit}&offset={next_offset}&filter={self.search_filter}`

We should only detect Yext if there is a possibility of extracting the required `api_key` value, otherwise the use of Yext is probably unrelated to finding stores, or is possibly in use behind the scenes but the brand website is using an API proxy to hide the key from the public.

Reply:
So, I'd be okay dropping the locator-google one.


The "answers search bar" detection I feel has a very strong correlation with being able to find API keys.
IE: https://www.fnb-online.com/atm-branch-locator + others
```
https://liveapi.yext.com/v2/accounts/me/answers/vertical/query?experienceKey=fnb-answers&api_key=10f82fdf7a37ee369b154241c59dade1&v=20190101&version=PRODUCTION&locale=en&input=&verticalKey=locations&limit=20&offset=0&retrieveFacets=true&facetFilters=%7B%7D&session_id=cb9a3c9e-9b5f-4ba7-b09a-f6b0480bdb16&sessionTrackingEnabled=true&sortBys=%5B%5D&referrerPageUrl=&source=STANDARD&jsLibVersion=v1.12.4
```
While this API endpoint is slightly, it does seem to present structured data that should be very similar to the entities endpoint.


window.Yext I have 2x examples of; and both were examples where an API call was subsequently made. However, that was existing storefinders I think.

Happy to make this `window.Yext` && some_further_pattern if you have examples where this gets a hit but doesn't give us anything useful

Reply:
From https://www.fnb-online.com/atm-branch-locator:

`<div class="desktop-search" id="search-box-19eafc2b374c463fa09cabc7b6739274" data-component="yext-search" data-apiKey="10f82fdf7a37ee369b154241c59dade1"`